iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/relational/sql/ast/AstMemoryEstimationHelper.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.db.queryengine.plan.relational.sql.ast;
+
+import org.apache.tsfile.utils.Accountable;
+import org.apache.tsfile.utils.RamUsageEstimator;
+
+import javax.annotation.Nullable;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Helper class for estimating memory usage of AST nodes. This class provides utility methods that
+ * can be used by Node subclasses to calculate their memory footprint.
+ */
+public final class AstMemoryEstimationHelper {
+
+  public static final long OPTIONAL_INSTANCE_SIZE =
+      RamUsageEstimator.shallowSizeOfInstance(Optional.class);
+
+  public static final long NODE_LOCATION_INSTANCE_SIZE =
+      RamUsageEstimator.shallowSizeOfInstance(NodeLocation.class);
+
+  private AstMemoryEstimationHelper() {
+    // hide the constructor
+  }
+
+  public static long getEstimatedSizeOfAccountableObject(@Nullable final Accountable accountable) {
+    return accountable == null ? 0 : accountable.ramBytesUsed();
+  }
+
+  public static long getEstimatedSizeOfString(@Nullable final String str) {
+    return str == null ? 0L : RamUsageEstimator.sizeOf(str);
+  }
```

Comment:
delete this method, let all callers directly call RamUsageEstimator.sizeOf(str).